app/src/main/java/com/spoony/spoony/core/network/ContentUriRequestBody.kt

```diff
@@ -0,0 +1,283 @@
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageDecoder
+import android.graphics.Matrix
+import android.media.ExifInterface
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import android.util.Size
+import androidx.annotation.RequiresApi
+import com.spoony.spoony.BuildConfig
+import java.io.ByteArrayOutputStream
+import javax.inject.Inject
+import kotlin.math.max
+import kotlin.math.min
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okio.BufferedSink
+import timber.log.Timber
+
+class ContentUriRequestBody @Inject constructor(
+    context: Context,
+    private val uri: Uri?,
+    private val config: ImageConfig = ImageConfig.DEFAULT
+) : RequestBody() {
+    private val contentResolver = context.contentResolver
+    private var compressedImage: ByteArray? = null
+    private var metadata: ImageMetadata? = null
+
+    private data class ImageMetadata(
+        val fileName: String,
+        val size: Long,
+        val mimeType: String?
+    ) {
+        companion object {
+            val EMPTY = ImageMetadata("", 0L, null)
+        }
+    }
+
+    data class ImageConfig(
+        val maxWidth: Int,
+        val maxHeight: Int,
+        val maxFileSize: Int,
+        val initialQuality: Int,
+        val minQuality: Int,
+        val format: Bitmap.CompressFormat = Bitmap.CompressFormat.JPEG
+    ) {
+        companion object {
+            val DEFAULT = ImageConfig(
+                maxWidth = 1024,
+                maxHeight = 1024,
+                maxFileSize = 1024 * 1024,
+                initialQuality = 100,
+                minQuality = 20
+            )
+        }
+    }
+
+    init {
+        uri?.let {
+            metadata = extractMetadata(it)
+        }
+    }
+
+    private fun extractMetadata(uri: Uri): ImageMetadata =
+        runCatching {
+            contentResolver.query(
+                uri,
+                arrayOf(
+                    MediaStore.Images.Media.SIZE,
+                    MediaStore.Images.Media.DISPLAY_NAME
+                ),
+                null,
+                null,
+                null
+            )?.use { cursor ->
+                if (cursor.moveToFirst()) {
+                    ImageMetadata(
+                        fileName = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DISPLAY_NAME)),
+                        size = cursor.getLong(cursor.getColumnIndexOrThrow(MediaStore.Images.Media.SIZE)),
+                        mimeType = contentResolver.getType(uri)
+                    )
+                } else {
+                    ImageMetadata.EMPTY
+                }
+            } ?: ImageMetadata.EMPTY
+        }.getOrDefault(ImageMetadata.EMPTY)
```

Comment:
p4) 위에 import하면 이렇게도 접근 가능하답니다~ 그런데 명시적으로 표시하기 위해 import하지 않는것도 좋은 방법이라 생각해서 참고만 하시길!

```suggestion
                    EMPTY
                }
            } ?: EMPTY
        }.getOrDefault(EMPTY)
```